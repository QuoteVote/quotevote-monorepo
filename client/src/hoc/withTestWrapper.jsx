import React from 'react'
import { PersistGate } from 'redux-persist/integration/react'
import { ApolloProvider } from '@apollo/react-hooks'
import { Provider } from 'react-redux'
import { createTheme, ThemeProvider } from '@material-ui/core/styles'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import store, { persistor } from '../store/store'
import client from '../config/apollo'
import customTheme from '../theme'
import { AuthModalProvider } from '../Context/AuthModalContext'
const hist = createBrowserHistory()
const theme = createTheme(customTheme)

const withTestWrapper = (Component) => (props) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <ThemeProvider theme={theme}>
          <AuthModalProvider>
            <Router history={hist}>
              <Component {...props} />
            </Router>
          </AuthModalProvider>
        </ThemeProvider>
      </PersistGate>
    </Provider>
  </ApolloProvider>
)
export default withTestWrapper
