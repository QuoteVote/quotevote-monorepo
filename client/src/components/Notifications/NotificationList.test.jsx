import React from 'react'
import { render, act } from '@testing-library/react'

// Component being tested
import NotificationLists from './NotificationLists'
import withTestWrapper from '../../hoc/withTestWrapper'

const NotificationListsWrapper = withTestWrapper(NotificationLists)

describe('NotificationLists test -', () => {
  it('renders correctly', async () => {
    let container
    await act(async () => {
      const res = render(
        <NotificationListsWrapper notifications={[]} />
      )
      container = res.container
    })
    expect(container.firstChild).toMatchSnapshot()
  })
})
