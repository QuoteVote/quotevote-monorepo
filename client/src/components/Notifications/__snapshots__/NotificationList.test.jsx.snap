// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotificationLists test - > renders correctly 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column makeStyles-rootNoNotification-3 makeStyles-rootNoNotification-9 css-1bkjq21-MuiGrid-root"
  justify="center"
>
  <div
    class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
  >
    <img
      alt=""
      src="/assets/ZeroNotificationsBG.png"
    />
  </div>
  <div
    class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
  >
    <span
      class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
    >
      Relax, you don't have any alerts right now.
    </span>
  </div>
</div>
`;
